De-duplicate case 12 (custom levels) in mapclass::loadlevel(), mostly entity creation

InfoTeddy opened the pull request:
Case 12 in `mapclass::loadlevel()` is a hot mess. A lot of it is a bunch of copy-pasted code, especially the part where edentities are converted into actual entities by a bunch of `obj.createentity()` calls. Luckily, I'm here to clean it all up and make it better.

## Legal Stuff:

By submitting this pull request, I confirm that...

- [X] My changes may be used in a future commercial release of VVVVVV (for
  example, a 2.3 update on Steam for Windows/macOS/Linux)
- [X] I will be credited in a `CONTRIBUTORS` file and the "GitHub Friends"
  section of the credits for all of said releases, but will NOT be compensated
  for these changes

flibitijibibo commented:
b1c0eed: This adds some more calculations done to all cases, rather than just the 2 cases where the bounding math was present.
512750f7: Same here, this does an assignment for cases that may not be those three values. We would need to assert that no other cases are possible, at which point we could probably inline most of the other statements in the switch block as well.

The other commits look okay to me.

InfoTeddy commented:
GitHub doesn't seem to work properly right now. I have rebased and force-pushed to resolve conflicts with #361 and address your issues (namely I dropped the `graphics.rcol` commit along with making sure the bounding logic is only calculated for enemies and moving platforms), but it doesn't seem to be updating for this PR.

flibitijibibo commented:
Rebased manually, again...
